include/proxsuite-nlp/prox-solver.hxx

```diff
@@ -228,7 +228,15 @@ template <typename Scalar>
 void ProxNLPSolverTpl<Scalar>::computeProblemDerivatives(
     const ConstVectorRef &x, Workspace &workspace, boost::mpl::true_) const {
   this->computeProblemDerivatives(x, workspace, boost::mpl::false_());
-  problem_->computeHessians(x, workspace, hess_approx == HessianApprox::EXACT);
+  if (hess_approx == HessianApprox::BFGS) {
+    bfgs_strategy_->update(x, workspace.objective_gradient);
+    workspace.objective_hessian = bfgs_strategy_->M;
```

Comment:
Maybe, `workspace.objective_hessian` should be a `bfgs_strategy_->update` parameter to avoid copy.
This also allow to initialize the hessian (don't start at identity).

Reply:
good point, i modified this